R/create.R

```diff
@@ -240,22 +240,27 @@ check_not_nested <- function(path, name) {
   invisible()
 }
 
-rationalize_fork <- function(fork, repo_info, pat_available, user = NULL) {
-
+rationalize_fork <- function(fork, repo_info, auth_token) {
   perms <- repo_info$permissions
   owner <- repo_info$owner$login
 
   if (is.na(fork)) {
-    fork <- pat_available && !isTRUE(perms$push)
+    fork <- have_github_token(auth_token) && !isTRUE(perms$push)
   }
 
-  if (fork && !pat_available) {
-    check_github_token(auth_token = NULL)
+  if (fork && !have_github_token(auth_token)) {
```

Comment:
I wonder if this logic would be clearer if the outer condition tested for the presence/absence of the `auth_token`